vizro-core/examples/scratch_dev/app.py

```diff
@@ -1,36 +1,41 @@
 """Dev app to try things out."""
 
-import pandas as pd
 import vizro.models as vm
 import vizro.plotly.express as px
 from vizro import Vizro
 from vizro.models.types import capture
 
-animals = pd.DataFrame(
-    {"animals": ["giraffes", "orangutans", "monkeys"], "value": [20, 14, 23], "color": ["Male", "Female", "Female"]}
-)
-
 iris = px.data.iris()
 
 
-@capture("graph")
-def bar(data_frame):
-    """LA LA LA LA."""
-    fig = px.bar(data_frame, x="sepal_length", y="sepal_width", color="species")
-    return fig
-
-
-@capture("graph")
-def px_bar(data_frame):
-    """LA LA LA LA."""
-    fig = px.bar(data_frame, x="animals", y="value", color="color")
-    return fig
+@capture("action")
+def update_graph_title(value):
+    """Custom action."""
+    return f"Distribution for {value}"
 
 
 page = vm.Page(
-    title="My first page",
+    title="Dynamic Graph title",
     components=[
-        vm.Graph(figure=bar(iris)),
+        vm.Graph(
```

Comment:
I agree that the custom action docs example with the timeout is not much useful. We added this one in the docs just to show how custom actions could be created without specifying any `inputs` or `outputs`. I also agree that we have to update actions examples in the docs, but I'd wait until actions v2 is merged because it brings many advantages (one of these enhancements is that your example could filter graph data_frame and additionally update its title).

Reply:
Do you guys have a rough time estimate when the docs for actions V2 will be updated? If this takes another 2-3 weeks then I would prefer updating it now already, given that it's just a quick replacement of example and we have the code for it already?

Reply:
Optimistically, the action v2 documents will be merged by the end of October, so I agree we should replace this "time.sleep" action example in the docs. 

However, I don't think that this example from the scratch folder should replace the "time.sleep" action example as this one uses action's "inputs"/"outputs" and we already have some docs examples for that case. Also, this scratch example can confuse our users as it doesn't filer the graph (but the vm.Filter model is used). 

If you find any better example that doesn't use action's inputs/outputs, I'm fine to replace the dummy "time.sleep" one. And, I'm also okay with completely deleting this example from the docs if we can't find any better example.

Reply:
I definitely see the point on the `vm.Filter`. I forgot that it replaces the action instead of appending.. 🤔 

On the improved example - why do we even want to show an example without input/output? Is this something we commonly do? In the Dash docs example, they also show a basic callback with input/output which I find more intuitive. While there is a current example with input/output, I find that one a bit more advanced. I was looking for a replicated example similar to the Dash docs where `XXX.value` affects `XXX.children`

Reply:
@huong-li-nguyen @antonymilne
If you think it fits, we can also adjust the simple "time.sleep" example to:

```py
@capture("action")
def mock_upload_downloaded_file_to_s3(mock_s3_url: str):
    """Mocked custom action that represents."""
    sleep(2)
   # Uploading downloaded file to s3
   # ...

...

        vm.Button(
            text="Export data",
            actions=[
                vm.Action(function=export_data()),
                vm.Action(
                    function=mock_upload_downloaded_file_to_s3(mock_s3_url="s3://my-bucket-name/<KEY>.csv")
                ),
                vm.Action(function=export_data(file_format="xlsx")),
            ]
        )
```

Reply:
Sounds good! 

I've (1) left the time.sleep example as is, (2) used the updated example from @antonymilne above and (3) simplified the existing input/output example with `clickData`. 

I think the configuration for that looked more complex than it needed to be. For example, it had a control, filter_interaction, and some more advanced layouts, which I stripped off now because I think the example was supposed to only show "how to create a custom action that shows the clickData of a chart in a Card." 

And for that purpose, I don't think we need the filter, the second chart, the chart interaction, etc.. But let me know if I should revert that change: https://github.com/mckinsey/vizro/pull/714/commits/501228821ca70a3a430a55cbfdc880ecf861f5a2

Reply:
Yeah, sounds good, let's do that 👍